datasets/mdsd/mdsd.py

```diff
@@ -0,0 +1,141 @@
+# coding=utf-8
+# Copyright 2020 HuggingFace Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+import os
+import datasets
+import xml
+from lxml import html
+import xmltodict
+
+_DESCRIPTION = """\
+The Multi-Domain Sentiment Dataset contains product reviews taken from Amazon.com from 4 product types (domains): Kitchen, Books, DVDs, and Electronics. Each domain has several thousand reviews, but the exact number varies by domain. Reviews contain star ratings (1 to 5 stars) that can be converted into binary labels if needed. This page contains some descriptions about the data.
+"""
+_HOMEPAGE_URL = ""
+_CITATION = None
+_URL = "http://www.cs.jhu.edu/~mdredze/datasets/sentiment/domain_sentiment_data.tar.gz"
+_MISSING_BOOKS_UNLABELED = (
+    "http://www.cs.jhu.edu/~mdredze/datasets/sentiment/book.unlabeled.gz"
+)
+_VERSION = "1.0.0"
+_DOMAINS = ["books", "dvd", "electronics", "kitchen_&_housewares"]
+_ALL = "all"
+
+
+class MDSDatasetConfig(datasets.BuilderConfig):
+    def __init__(self, *args, domains=None, **kwargs):
+        super().__init__(*args, version=datasets.Version(_VERSION, ""), **kwargs)
+        self.domains = domains
+
+
+class MDSDataset(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class MDSD(datasets.GeneratorBasedBuilder):
```
must match the dataset script name

You'll need to regenerate the dataset_infos.json after this change